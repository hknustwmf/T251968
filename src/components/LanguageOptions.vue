<template>
	<div class="filter">
		<label for="autocomplete-dropdown">Select a language:</label>
		<autocomplete-dropdown
			id="autocomplete-dropdown"
			:options="languageOptions"
			v-model="selectedLanguage"
		/>
	</div>
</template>

<script>
import availableLanguages from '@/langs';
import AutocompleteDropdown from '@/components/AutocompleteDropdown';

export default {
	components: {
		'autocomplete-dropdown': AutocompleteDropdown,
	},
	data() {
		return {
			selectedLanguage: '',
			languageOptions: [],
		};
	},
	methods: {
		getLanguageList() {
			this.languageOptions = availableLanguages.languages;
		},
	},
	mounted() {
		this.getLanguageList();
	},
};
</script>

<style scoped>
.filters {
	width: 800px;
	margin: 0 auto;
	border: 2px solid;
}

.filter {
	text-align: left;
}

.result {
	margin-top: 30px;
	text-align: left;
}

label {
	display: block;
}
</style>
