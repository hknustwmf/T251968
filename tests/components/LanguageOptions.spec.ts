import LanguageOptions from '@/components/LanguageOptions.vue';
import { shallowMount } from '@vue/test-utils';

describe( 'LanguageOptions', () => {
	it( 'renders correctly', () => {
		const wrapper = shallowMount( LanguageOptions );
		expect( wrapper.element ).toMatchSnapshot();
	} );

	it( 'mounted correctly', () => {
		const wrapper = shallowMount( LanguageOptions );
		expect( wrapper.isVueInstance() ).toBe( true );
	} );
} );
