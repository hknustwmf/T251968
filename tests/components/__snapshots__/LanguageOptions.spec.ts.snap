// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageOptions renders correctly 1`] = `
<div
  class="filter"
>
  <label
    for="autocomplete-dropdown"
  >
    Select a language:
  </label>

  <autocomplete-dropdown-stub
    id="autocomplete-dropdown"
    options=""
    value=""
  />
</div>
`;
